ext/mbstring/tests/cp1252_encoding.phpt

```diff
@@ -0,0 +1,57 @@
+--TEST--
+Exhaustive test of verification and conversion of CP1252 text
+--SKIPIF--
+<?php
+extension_loaded('mbstring') or die('skip mbstring not available');
+if (getenv("SKIP_SLOW_TESTS")) die("skip slow test");
+?>
+--FILE--
+<?php
+srand(789); // Make results consistent
+include('encoding_tests.inc');
+mb_substitute_character(0x25); // '%'
+
+function testValid($from, $to) {
+    testValidString($from, $to, 'CP1252', 'UTF-16BE');
+}
+function testInvalid($from, $to) {
+    testInvalidString($from, $to, 'CP1252', 'UTF-16BE');
+}
+
+$toUnicode = array();
+$fromUnicode = array();
+
+$fp = fopen(realpath(__DIR__ . "/data/CP1252.txt"), 'r+');
+while ($line = fgets($fp, 256)) {
+    if ($line[0] == '#')
+        continue;
+
+    if (sscanf($line, "0x%x\t0x%x", $byte, $codepoint) == 2) {
+        $toUnicode[$byte] = pack('n', $codepoint);
+        $fromUnicode[$codepoint] = chr($byte);
+    }
+}
+
+// Try all CP1252 bytes
+for ($i = 0; $i < 256; $i++) {
+    if (isset($toUnicode[$i])) {
+        testValid(chr($i), $toUnicode[$i]);
+    } else {
+        testInvalid(chr($i), "\x00%");
+    }
+}
+
+// Try all Unicode codepoints up to 0xFFFF
+for ($i = 0; $i <= 0xFFFF; $i++) {
+    if (isset($fromUnicode[$i])) {
+        testValid($fromUnicode[$i], pack('n', $i));
+    } else if (($i & 0xFC00) != 0xD800) {
+        convertInvalidString(pack('n', $i), "%", 'UTF-16BE', 'CP1252');
+    }
+}
+
+echo "All is fine and dandy for CP1252 text\n";
```

Comment:
The logic here looks the same as in the ISO-8859 test, and also the same as the CP1251 tests, and probably more. Maybe extract a function that just accepts the encoding name and the data file and works for all these simple encodings?

Reply:
That's a good idea.

Take it as done for all the single-byte encodings.